orchagent/fdborch.cpp

```diff
@@ -36,6 +37,73 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
     Orch::addExecutor(fdbNotifier);
 }
 
+bool FdbOrch::bake()
+{
+    Orch::bake();
+
+    auto consumer = dynamic_cast<Consumer *>(getExecutor(APP_FDB_TABLE_NAME));
+    if (consumer == NULL)
+    {
+        SWSS_LOG_ERROR("No consumer %s in Orch", APP_FDB_TABLE_NAME);
+        return false;
+    }
+
+    size_t refilled = consumer->refillToSync(&m_fdbStateTable);
+    SWSS_LOG_NOTICE("Add warm input FDB State: %s, %zd", APP_FDB_TABLE_NAME, refilled);
+    return true;
+}
+
+bool FdbOrch::storeFdbEntryState(const FdbUpdate& update)
+{
+    const FdbEntry& entry = update.entry;
+    const Port& port = update.port;
+    sai_vlan_id_t vlan_id = port.m_port_vlan_id;
+    const MacAddress& mac = entry.mac;
+    string portName = port.m_alias;
+
+    // ref: https://github.com/Azure/sonic-swss/blob/master/doc/swss-schema.md#fdb_table
+    string key = "Vlan" + to_string(vlan_id) + ":" + mac.to_string();
+
+    if (update.add)
+    {
+        auto inserted = m_entries.insert(entry);
+
+        SWSS_LOG_DEBUG("FdbOrch notification: mac %s was inserted into bv_id 0x%lx",
+                        entry.mac.to_string().c_str(), entry.bv_id);
+
+        if (!inserted.second)
+        {
+            SWSS_LOG_INFO("FdbOrch notification: mac %s is duplicate", entry.mac.to_string().c_str());
+            return false;
+        }
+
+        // Write to StateDb
+        std::vector<FieldValueTuple> fvs;
+        fvs.push_back(FieldValueTuple("port", portName));
+        fvs.push_back(FieldValueTuple("type", "dynamic"));
+        m_fdbStateTable.set(key, fvs);
```

Comment:
If FDB entries are saved to stateDB. During system warm reboot,  will stateDB be restored?

Reply:
Do you mean will stateDB in redis persistent during warm reboot? Yes, it should be.

Or do you mean if fdborch will restore from stateDB? Yes, restored in bake().

---
In reply to: [218964895](https://github.com/Azure/sonic-swss/pull/615#discussion_r218964895) [](ancestors = 218964895)

Reply:
no stateDB will be cleared during state warm reboot. it is a problem.

Reply:
I will save the fdb entries in state db in the warm-reboot script and recover it in the going up path. @jipanyang , thanks for the catch.

Reply:
Would it be easier if the data is put in a separate table in appDB so no special handling is needed for stateDB?

Reply:
Would it be easier if the data is put in a separate table in appDB so no special handling is needed for stateDB?